esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -164,6 +164,8 @@ void EsdlServiceImpl::init(const IPropertyTree *cfg,
     StringBuffer xpath;
     xpath.appendf("Software/EspProcess[@name=\"%s\"]/EspService[@name=\"%s\"]", process, service);
     IPropertyTree * srvcfg = cfg->queryPropTree(xpath.str());
+    xpath.setf("Software/EspProcess[@name=\"%s\"]/EspBinding[@service=\"%s\"]", process, service); //get this service's binding cfg
+    m_oEspBindingCfg.setown(cfg->queryPropTree(xpath.str()));
```

Comment:
Is this ultimately pointing directly to a remote DALI Ptree?  If so I wonder if there are any locking issues, and whether we should make a copy of the tree rather think linking to it.

Also you are doing a queryPropTree which doesn't link and then assingin ownership via setown.  That could end up causing a fault.

Reply:
I don't think so, CEspConfig creates the tree from the configfile and passes it down